includes/snmp.inc.php

```diff
@@ -651,6 +655,9 @@ function snmp_gen_auth(&$device)
     } elseif ($device['snmpver'] === 'v2c' or $device['snmpver'] === 'v1') {
         $cmd  = " -".$device['snmpver'];
         $cmd .= " -c '".$device['community']."'";
+        if ($vlan) {
+            $cmd .= '@' . $vlan;
```

Comment:
snmp v3 has a separate switch for context
snmp v2 and v1?  use the `@context` format.

I can implement that for you if you like.